components/list/src/lib.rs

```diff
@@ -12,13 +12,14 @@
 //! ## Formatting *and* lists in Spanish
 //!
 //! ```
-//! # use icu::list::{ListFormatter, ListLength};
+//! # use icu::list::{ListFormatter, ListFormatterOptions, ListLength};
 //! # use icu::locale::locale;
 //! # use writeable::*;
 //! #
 //! let list_formatter = ListFormatter::try_new_and_with_length(
-//!     &locale!("es").into(),
-//!     ListLength::Wide,
+//!     locale!("es").into(),
+//!     ListFormatterOptions::new()
+//!       .with_length(ListLength::Wide)
 //! )
```

Comment:
+1 

It would have been nice to make options changes in a separate PR, because those changes are most of the friction of this review. I'm aligned on the preferences stuff.

Reply:
I prefer to complete the migration of a single module to evaluate the DX.